assets/src/edit-story/components/panels/stylePreset/presetGroup.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import { useLayoutEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useKeyDownEffect } from '../../keyboard';
+
+const COLORS_PER_ROW = 6;
+const STYLES_PER_ROW = 3;
+const PRESET_HEIGHT = 35;
+
+const Group = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+`;
+
+const ButtonWrapper = styled.div`
+  flex-basis: ${({ width }) => width}%;
+  height: ${PRESET_HEIGHT}px;
+`;
+
+const Label = styled.div`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  font-size: 10px;
+  line-height: 12px;
+  text-transform: uppercase;
+  padding: 6px 0;
+`;
+
+function PresetGroup({ presets, itemRenderer, type, label }) {
+  const [activeIndex, setActiveIndex] = useState(0);
+  const groupRef = useRef(null);
+  const initialRender = useRef(true);
+
+  const getIndexDiff = (key, rowLength) => {
+    switch (key) {
+      case 'ArrowUp':
+        return -rowLength;
+      case 'ArrowDown':
+        return rowLength;
+      case 'ArrowLeft':
+        return -1;
+      case 'ArrowRight':
+        return 1;
+      default:
+        return 0;
+    }
+  };
+
+  useLayoutEffect(() => {
+    if (initialRender.current) {
+      initialRender.current = false;
+    } else if (groupRef.current) {
+      groupRef.current.querySelector('[tabindex="0"]').focus();
```

Comment:
Which element/component is this query expected to return? Could you please also comment?

Reply:
When the user uses arrow keys to navigate among the color presets, the active index will indicate which preset should be focused on. `tabIndex` is `0` only for the preset with index matching the `activeIndex`. Added comments, too.

Reply:
I think i got it. So all others will have `tabindex=-1`?

Reply:
Correct, all the others will have `tabIndex=-1` -- will add this to the comment, too.